plugins/catalog/src/components/CatalogKindHeader/CatalogKindHeader.tsx

```diff
@@ -61,6 +61,14 @@ export function CatalogKindHeader(props: CatalogKindHeaderProps) {
   });
   const { updateFilters, queryParameters } = useEntityList();
 
+  const queryParamKind = useMemo(
```

Comment:
I'd leave the `initialFilter` out of this `useMemo` so it's strictly concerned with the queryParameters as advertised by the variable name. You can leave off the lowercasing since the backend handles it. Then it should be simply:

```ts
const queryParamKind = useMemo(() => [queryParameters.kind].flat()[0], [queryParameters]);
const [selectedKind, setSelectedKind] = useState(queryParamKind ?? initialFilter);

// No need for a conditional here, since we're guaranteed to have a value
useEffect(() => setSelectedKind(queryParamKind), [queryParamKind]);
```

There's an opportunity to be more specific with the queryParam we care about too, I'm realizing I didn't do this in the other components but you could target `kind` specifically:

```ts
const { updateFilters, queryParameters: { kind: kindParameter }} = useEntityList();
const queryParamKind = useMemo(() => [kindParameter].flat()[0], [kindParameter]);
```
That will avoid firing the `useMemo` when other unrelated query params change.

Reply:
Sweet! I've made the changes you've suggested, thanks 😄